src/Adapter/MSTest.TestAdapter/MSTestSettings.cs

```diff
@@ -262,32 +264,49 @@ public static void PopulateSettings(MSTestSettings settings)
     /// <param name="logger"> The logger for messages. </param>
     /// The discovery context that contains the runsettings.
     /// </param>
-    internal static void PopulateSettings(IDiscoveryContext? context, IMessageLogger? logger)
+    internal static void PopulateSettings(IDiscoveryContext? context, IMessageLogger? logger, IConfiguration? configuration)
     {
-        RunConfigurationSettings = RunConfigurationSettings.PopulateSettings(context);
-
-        if (context?.RunSettings == null || StringEx.IsNullOrEmpty(context.RunSettings.SettingsXml))
+        if (configuration != null && context?.RunSettings != null)
         {
-            // This will contain default adapter settings
-            CurrentSettings = new MSTestSettings();
-            return;
+            throw new InvalidOperationException(Resource.DuplicateConfigurationError);
         }
 
-        MSTestSettings? aliasSettings = GetSettings(context.RunSettings.SettingsXml, SettingsNameAlias, logger);
-
-        // If a user specifies MSTestV2 in the runsettings, then prefer that over the v1 settings.
-        if (aliasSettings != null)
+        if (context?.RunSettings != null && !StringEx.IsNullOrEmpty(context.RunSettings.SettingsXml))
         {
-            CurrentSettings = aliasSettings;
+            RunConfigurationSettings = RunConfigurationSettings.PopulateSettings(context);
+
+            MSTestSettings? aliasSettings = GetSettings(context.RunSettings.SettingsXml, SettingsNameAlias, logger);
+
+            // If a user specifies MSTestV2 in the runsettings, then prefer that over the v1 settings.
+            if (aliasSettings != null)
+            {
+                CurrentSettings = aliasSettings;
+            }
+            else
+            {
+                MSTestSettings? settings = GetSettings(context.RunSettings.SettingsXml, SettingsName, logger);
+
+                CurrentSettings = settings ?? new MSTestSettings();
+            }
+
+            SetGlobalSettings(context.RunSettings.SettingsXml, CurrentSettings, logger);
+            return;
         }
-        else
+
+        if (configuration is not null)
         {
-            MSTestSettings? settings = GetSettings(context.RunSettings.SettingsXml, SettingsName, logger);
+            RunConfigurationSettings = RunConfigurationSettings.PopulateSettings(configuration);
+
+            MSTestSettings? settings = GetSettingsFromConfig(configuration, logger);
 
             CurrentSettings = settings ?? new MSTestSettings();
+
+            return;
         }
 
-        SetGlobalSettings(context.RunSettings.SettingsXml, CurrentSettings, logger);
+        // This will contain default adapter settings
+        CurrentSettings = new MSTestSettings();
+        return;
```

Comment:
not needed right?

Reply:
it's needed if nor runsettings or config files are there o set the default settings